tests/test_dfes.py

```diff
@@ -0,0 +1,776 @@
+        with pytest.raises(ValueError) as e_info:
+            dfe.MutationType(
+                distribution_type="l",
+                distribution_args=[1, 2, 3, 4],
+            )
+        assert (
+            str(e_info.value)
+            == "Lognormally-distributed sel. coefs. (distribution_type='l') "
+            "use a (meanlog, sdlog) parameterisation, requiring sdlog > 0."
+        )
+
+        with pytest.raises(ValueError) as e_info:
+            dfe.MutationType(
+                distribution_type="l",
+                distribution_args=[1, -2],
+            )
+        assert str(e_info.value) == "The sdlog parameter must be nonnegative."
+
+    def test_mutation_type_is_neutral(self):
+        mt = dfe.MutationType()
+        assert mt.is_neutral is True
+
+        mt = dfe.MutationType(
+            distribution_type="g",
+            distribution_args=[0.014, 0.19],
+        )
+        assert mt.is_neutral is False
+
+    def test_mutation_types(self):
+        mut_params = {
+            "f": ([-0.1], [0], [0.1], [50]),
+            "g": ([-0.1, 0.1], [0.1, 0.1], [50, 50]),
+            "e": ([0.1], [10], [5000], [0]),
+            "n": ([-0.1, 0.2], [0.1, 0.1], [50, 50]),
+            "w": ([0.1, 0.2], [0.1, 0.1], [50, 50]),
+            "l": ([-0.1, 0.2], [0.1, 0.1], [50, 50]),
+        }
+        for t in mut_params:
+            for p in mut_params[t]:
+                mt = dfe.MutationType(distribution_type=t, distribution_args=p)
+                if t == "l":
+                    assert mt.distribution_type == "s"
+                else:
+                    assert mt.distribution_type == t
+                    assert len(mt.distribution_args) == len(p)
+                    for a, b in zip(mt.distribution_args, p):
+                        assert a == b
+
+    def test_bad_mutation_types(self):
+        bad_mut_params = {
+            "f": ([0.1, 0.2], [], [np.inf]),
+            "g": ([], [0.1, 0], [0.1, -0.1], [0.1, 0.4, 0.5], [0.1, np.inf]),
+            "e": ([], [0, 1], [0.1, 0.4, 0.5], [np.inf]),
+            "n": ([], [0.1, -1], [0.1, 0.4, 0.5], [0.1], [0.3, np.inf]),
+            "w": ([], [-0.1, 1], [0.1, -1], [0.1, 0.4, 0.5], [0.1], [np.inf, 2.3]),
+            "l": ([], [0.1, -1], [0.1, 0.4, 0.5], [0.1], [0.1, np.inf]),
+        }
+        for t in bad_mut_params:
+            for p in bad_mut_params[t]:
+                print(t, p)
+                with pytest.raises(ValueError):
+                    dfe.MutationType(distribution_type=t, distribution_args=p)
+
+    def test_convert_to_substitution(self):
+        mt = dfe.MutationType()
+        assert mt.convert_to_substitution is True
+        for c in (True, False):
+            mt = dfe.MutationType(convert_to_substitution=c)
+            assert mt.convert_to_substitution == c
+
+    def test_dominance_coeff(self):
+        mt = dfe.MutationType()
+        assert mt.dominance_coeff == 0.5
+        for dominance_coeff in (-10, 0, 0.5, 1, 50):
```

Comment:
@petrelharp because I found the test codes allows `dominance_coeff` to be `int`. So we should change this line into `for dominance_coeff in (-10.0, 0.0, 0.5, 1.0, 50.0)` and only allow `dominance_coeff` as `float`?

Reply:
If you pass in an int to dominance_coeff do we get an error? We could put a `float( )` around `dominance_coeff`?

Reply:
No error now, because I changed the condition from `if not isinstance(self.dominance_coeff, float):` to `if not isinstance(self.dominance_coeff, (float, int)):`

Reply:
No harm in saying "float" in the docs, though.